### PR TITLE
Client/server: Communicate location solely using S2 cell ids at L9

### LINKED ISSUE #1040: Client shall send and server shall record solely a L9 S2 cell id
- for v1: neither client nor server will geocode nor rev geocode anything
- for v1: client will get location permission (“coarse” on Android, which is about 1 city block) then quantify that to (L9) S2 cell ids of no less than (10km)^2 and transmit solely that id. 
- for v1: no code will be added to track users over time even if their location changes substantially.
- in the future: users can choose their location.
- in the future: if/when we need more precise GPS location-dependent features we can build it at that time.

### PR DESCRIPTION
**Pull request by advayDev1**
- Removes all geocoding and its dependencies
- Switches Android permissions to coarse location, and asks at runtime for Low accuracy iOS location
- Quantizes location to S2CellId at L9 (strictly greater than 100km^2)
- Removes all other fields of location communication
- Server-side validation of s2 token being of a coarse-enough level to avoid pollution and for privacy
- New dependency: Have reviewed the specific pinned version from pub.dev.

TESTED=true with iOS and Android simulator against Dev app engine backend

Close #1040

### CODE REVIEW

**Comment by SamMousa** [comment]
Did you see there is a fork that is more actively maintained (but probably used less)?

**Comment by advayDev1** [comment]
No - its a direct port and I think that's all that matters.

File: api/who/who.proto [hunk 1]
```diff
@@ -21,17 +21,10 @@ message PutDeviceTokenRequest {
...
-  string adminArea1 = 5;
-  string adminArea2 = 6;
-  string adminArea3 = 7;
-  string adminArea4 = 8;
-  string adminArea5 = 9;
-  string locality = 10;
+  // A token is a hex representation of the 64-bit S2CellId
+  // in string format that will also survive JSON (which
+  // lacks 64-bit number precision).
+  string s2CellIdToken = 11;
```
Comment (advayDev1) [inline api/who/who.proto:27]:
```suggestion
  // The level (specificity) of the S2CellId must not exceed
  // who.Client.MAX_S2_CELL_LEVEL.
  string s2CellIdToken = 11;
```

**Comment by advayDev1** [comment]
Thanks Bob!  @patniemeyer how's this flutter-wise?

File: api/who/who.proto [hunk 2]
```diff
@@ -21,17 +21,12 @@ message PutDeviceTokenRequest {
...
-  string adminArea3 = 7;
-  string adminArea4 = 8;
-  string adminArea5 = 9;
-  string locality = 10;
+  // A token is a hex representation of the 64-bit S2CellId
+  // in string format that will also survive JSON (which
+  // lacks 64-bit number precision).
+  // The level (specificity) of the S2CellId must not exceed
+  // who.Client.MAX_S2_CELL_LEVEL.
+  string s2CellIdToken = 11;
```
Comment (SamMousa) [inline api/who/who.proto:29]: Shouldn't the field number for this field be 1?

Comment (advayDev1) [reply api/who/who.proto:29]: I'm sticking with protobuf best practice of never re-using a number.

**Pull request merged**